pkgs/by-name/ll/llm-ls/package.nix

```diff
@@ -19,21 +24,38 @@ rustPlatform.buildRustPackage {
     sha256 = "sha256-ICMM2kqrHFlKt2/jmE4gum1Eb32afTJkT3IRoqcjJJ8=";
   };
 
-  cargoHash = "sha256-Fat67JxTYIkxkdwGNAyTfnuLt8ofUGVJ2609sbn1frU=";
+  cargoHash = "sha256-m/w9aJZCCh1rgnHlkGQD/pUDoWn2/WRVt5X4pFx9nC4=";
+
+  cargoPatches = [
+    # https://github.com/huggingface/llm-ls/pull/102
+    ./fix-time-compilation-failure.patch
+    # https://github.com/huggingface/llm-ls/commit/479401f3a5173f2917a888c8068f84e29b7dceed
+    (fetchpatch {
+      name = "fix-version.patch";
+      url = "https://github.com/huggingface/llm-ls/commit/479401f3a5173f2917a888c8068f84e29b7dceed.patch?full_index=1";
+      hash = "sha256-4gXasfMqlrrP8II+FiV/qGfO7a9qFkDQMiax7yEua5E=";
+    })
+  ];
 
   buildAndTestSubdir = "crates/llm-ls";
 
   nativeBuildInputs = [ pkg-config ];
 
-  buildInputs = [ openssl ];
+  buildInputs = lib.optionals stdenv.isDarwin [
+    libiconv
```

Comment:
libiconv should usually be unconditional — it's a no-op on platforms that have it built in like Glibc, but if a package needs libiconv, it'll probably need it on all platforms that don't have it as part of the libc package.